src/main/java/stsjorbsmod/powers/MirrorPower.java

```diff
@@ -0,0 +1,58 @@
+package stsjorbsmod.powers;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DamageAction;
+import com.megacrit.cardcrawl.actions.common.ReducePowerAction;
+import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+
+public class MirrorPower extends CustomJorbsModPower {
+    public static final StaticPowerInfo STATIC = StaticPowerInfo.Load(MirrorPower.class);
+    public static final String POWER_ID = STATIC.ID;
+
+    public MirrorPower(final AbstractCreature owner, final int numberTurns) {
+        super(STATIC);
+
+        this.owner = owner;
+        this.amount = numberTurns;
+        this.isTurnBased = true;
+
+        updateDescription();
+    }
+
+    @Override
+    public void atStartOfTurn() {
+        if (this.amount == 1) {
+            AbstractDungeon.actionManager.addToBottom(new RemoveSpecificPowerAction(this.owner, this.owner, this.ID));
+        } else {
+            AbstractDungeon.actionManager.addToBottom(new ReducePowerAction(this.owner, this.owner, this.ID, 1));
+        }
+    }
+
+    @Override
+    public int onAttacked(DamageInfo info, int damageAmount) {
```

Comment:
When we needed a similar effect for Mirror Image, I got feedback that it was weird for the attack animation to still show on the player when it was being redirected; you can avoid that by using `OnDamageToRedirectSubscriber` like Mirror Image does, eg:

```java
    @Override
    public boolean onDamageToRedirect(AbstractPlayer player, DamageInfo info, AttackEffect attackEffect) {
        if (info.type != DamageInfo.DamageType.THORNS && info.type != DamageInfo.DamageType.HP_LOSS && info.owner != null && info.owner != this.owner) {
            this.flash();
            this.addToTop(new DamageAction(info.owner, new DamageInfo(this.owner, info.output, DamageInfo.DamageType.NORMAL), attackEffect, false));
            return true;
        }
        return false;
    }
```